cpu/samd21/cpu.c

```diff
@@ -204,8 +204,18 @@ static void clk_init(void)
                          | SYSCTRL_XOSC32K_ENABLE;
 #endif
 
+    /* Setup Clock generator 3 with divider 4 (1024 Hz) */
+    GCLK->GENDIV.reg  = (GCLK_GENDIV_ID(3)  | GCLK_GENDIV_DIV(4));
+    GCLK->GENCTRL.reg = (GCLK_GENCTRL_ID(3) | GCLK_GENCTRL_GENEN
+                      | GCLK_GENCTRL_RUNSTDBY | GCLK_GENCTRL_DIVSEL
+#if GEN2_ULP32K
+                      | GCLK_GENCTRL_SRC_OSCULP32K);
+#else
+                      | GCLK_GENCTRL_SRC_XOSC32K);
+#endif
+
     /* redirect all peripherals to a disabled clock generator (7) by default */
-    for (int i = 0x3; i <= 0x22; i++) {
+    for (int i = 0x4; i <= 0x22; i++) {
```

Comment:
I don't think the 0x3 (now 0x4) has anything to do with number of the clock generator. The 0x3 is most likely `GCLK_CLKCTRL_ID_WDT_Val`. Have a look at `cpu/sam0_common/include/vendor/samd21/include/component/gclk.h`.

This for-loop is a bit puzzling for me. I think we are trying to set all remaining(?) peripherals to use GCLK7.

We shouldn't be using "magic numbers". 0x22 is probably incorrect too. I see `GCLK_CLKCTRL_ID_I2S_1_Val` which is 0x24. Mind you the datasheet and the vendor include files aren't consistent either.

Reply:
I think that loop is pretty bogus to begin with.
I would suspect there is no change in current consumption when simply removing it - peripherals are not randomly clocked after reset.

But I'd rather make a separate PR for that.

Reply:
OK